Research/lva-node-app/public/js/main.js

```diff
@@ -0,0 +1,638 @@
+var cameras={};
+
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// universal functions
+
+/**
+* get global variables from server, Returns a promise so that functions that depend on these variables can wait
+* @returns {Promise<any>}
+*/
+function getGlobals()
+{
+  return new Promise((resolve, reject) => 
+  {
+    var request = sendRequest("", `http://localhost:${PORT}/getglobals`, "GET");
+    request.onreadystatechange = function () 
+    {
+      if(request.readyState == 4 && request.status == 200)
+      {
+        let response= JSON.parse(request.response);
+        console.log(response);
+        for (var list in response)
+        {
+          let temp=response[list].name;
+          switch(temp)
+          {
+            case 'graphInstances':
+              graphInstances=response[list].value;
+              break;
+            case 'graphTopologies':
+              graphTopologies=response[list].value;
+              break;
+            case 'cameras':
+              cameras=response[list].value;
+              break;
+          }
+        }
+        resolve(response);
+      }
+    }
+  });
+}
+
+/**
+* send current values to the server to be stored for later 
+*/
+function sendGlobals()
+{
+  let globals = 
+  [
+    {
+      "name": "graphInstances",
+    "value": graphInstances
+    },
+    {
+      "name": "graphTopologies",
+      "value": graphTopologies
+    },
+    {
+      "name": "cameras",
+      "value": cameras
+    }
+  ]
+  sendRequest(globals, `http://localhost:${PORT}/globals`);
+}
+
+/**
+*  send request with given parameters to a defined url in server. default method is POST
+* @param {jsonObject} payload - json object to be sent in request to server
+* @param {string} url - @link to communicate on
+* @param {string} requestType - POST or GET request 
+* @returns {XMLHttpRequest} - generated request
+*/ 
+function sendRequest(payload, url, requestType="POST") 
+{
+  let request = new XMLHttpRequest();
+  request.open(requestType, url, true);
+  request.setRequestHeader('Content-Type', 'application/json; charset=UTF-8');
+  if(requestType=="GET")
+  {
+    request.send();
+  }
+  else
+  {
+    request.send(JSON.stringify(payload));
+  }  
+  return request;
+}
+
+/**
+* delete all htmlElementents nested within the nearest parent li item
+* @param {HTMLButtonElement} htmlElement - what user clicked on
+*/
+function deleteFromParentListItem(htmlElement) 
+{
+  let id = $(htmlElement).closest('li').attr('id');
+  let item = document.getElementById(id);
+  item.parentElement.removeChild(item);
+}
+
+/**
+* create unique ID for an item
+* @param {string} nameToInclude - optional name to include in unique id
+* @returns {string} - unique ID
+*/
+function makeUniqueId(nameToInclude = "") 
+{
+  return (nameToInclude + CreateUUID());
+}
+
+/**
+* Creates a UUID 
+* @returns {UUID} - unique identifier
+*/
+function CreateUUID() 
```

Comment:
looks like this is only called from just above. doesn't seem worth being a separate method.

Reply:
Changed!